### PR TITLE
Recipe app- Ruby on Rails

### PR DESCRIPTION
**Pull request by myelin0**
### Here is our final pull request for the recipe app project. The project requirements we followed are the below:

- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
    - Should be built with Devise.
- Food list:
    - Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
    - Should lead to a form that allows users to add new food.
- Recipes list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
    - Should display a list of all public recipes ordered by newest as in the wireframe.
    - Should lead to recipe details.
- Recipe details:
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view 
    - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
    - Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.
- **Technical setup**
- Set up the repository on GitHub and use Gitflow.
- Set up Devise for authentication.
- Set up RSpec and Capybara testing libraries.

### CODE REVIEW

File: app/views/foods/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<div class="container w-full p-4 b-2">
+    <h2 class="text mb-4">Add new food</h2>
+    <%= form_with model: @food, url: foods_path,  local: true, data: {turbo:false} do |f| %>
+        <div class="mb-3">
+            <%= f.label :name, class: "form-label" %>
+            <%= f.text_field :name, class: "form-control" %>
+        </div>
+        <div class="mb-3">
+            <%= f.label :measurement_unit, class: "form-label" %>
+            <%= f.text_field :measurement_unit, class: "form-control" %>
```
Comment (thecodechaser) [inline app/views/foods/new.html.erb:10]: - [x] When I'm trying to create food, I'm getting this error. Please make sure to fix this issue.
image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @myelin0, and @Graycemuthui👋, 

### You are doing a marvelous job on the RoR group project and this is one of the best projects I have seen so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Awesome UI design. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/foods/new.html.erb (hunk 1 above)
Comment (myelin0) [reply app/views/foods/new.html.erb:10]: Hello @thecodechaser! :wave: Thank you so much for your review.
The schema has a "user_id" foreign key in the foods table. I guess the migration file for it was deleted along the way. I recreated a migration to add user_id in the Foods table. Thanks for your review!:raised_hands:
image

**Review by KDlamini — APPROVED** [review]
### Hi team @myelin0, @Graycemuthui 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Great PR title and description
✔️ Models, controllers, views, and routes setup correctly
✔️ User authentication implemented properly
✔️ No linter errors

# Status: Approved ✔️ 

simon

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**